src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/LanguageServerOperationContext.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Intellisense;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Handlers
+{
+    /// <summary>
+    /// A context for a language server operation with information that might be needed when handling the operation.
+    /// It also has helper methods to simplify the operation handling.
+    /// </summary>
+    public class LanguageServerOperationContext

```

Comment:
>LanguageServerOperationContext [](http://example.com/codeflow?start=17&length=30)

Can we document the lifespan of this? I assume this is per-request (given the requestId) , and not something that can be cached.